cs/src/core/Index/FasterLog/FasterLog.cs

```diff
@@ -368,6 +425,10 @@ private void Commit(long flushAddress)
                     CommittedUntilAddress = flushAddress;
                     // info.DebugPrint();
                 }
+
+                var _commitTask = commitTask;
+                commitTask = new TaskCompletionSource<long>();
+                _commitTask.SetResult(flushAddress);
```

Comment:
is this SetResult guaranteed to happen only once since commit's are done on a single background thread?
If not, a second (accidental) call to SetResult will result in an InvalidOperationException.

Reply:
Since SetResult happens under a lock (per commit), we should be fine.